src/plugins/fs/exfat.c

```diff
@@ -0,0 +1,302 @@
+#define DEPS_DUMPEXFAT_MASK (1 << DEPS_DUMPEXFAT)
+
+#define DEPS_LAST 5
+
+static const UtilDep deps[DEPS_LAST] = {
+    {"mkexfatfs", NULL, NULL, NULL},
+    {"exfatfsck", NULL, NULL, NULL},
+    {"exfatlabel", NULL, NULL, NULL},
+    {"dumpexfat", NULL, NULL, NULL},
+};
+
+static guint32 fs_mode_util[BD_FS_MODE_LAST+1] = {
+    DEPS_MKEXFAT_MASK,      /* mkfs */
+    0,                      /* wipe */
+    DEPS_EXFATFSCK_MASK,    /* check */
+    DEPS_EXFATFSCK_MASK,    /* repair */
+    DEPS_EXFATLABEL_MASK,   /* set-label */
+    DEPS_DUMPEXFAT_MASK,    /* query */
+    0,                      /* resize */
+    0,                      /* set-uuid */
+};
+
+#define UNUSED __attribute__((unused))
+
+/**
+ * bd_fs_exfat_is_tech_avail:
+ * @tech: the queried tech
+ * @mode: a bit mask of queried modes of operation (#BDFSTechMode) for @tech
+ * @error: (out): place to store error (details about why the @tech-@mode combination is not available)
+ *
+ * Returns: whether the @tech-@mode combination is available -- supported by the
+ *          plugin implementation and having all the runtime dependencies available
+ */
+gboolean __attribute__ ((visibility ("hidden")))
+bd_fs_exfat_is_tech_avail (BDFSTech tech UNUSED, guint64 mode, GError **error) {
+    guint32 required = 0;
+    guint i = 0;
+    for (i = 0; i <= BD_FS_MODE_LAST; i++)
+        if (mode & (1 << i))
+            required |= fs_mode_util[i];
+
+    return check_deps (&avail_deps, required, deps, DEPS_LAST, &deps_check_lock, error);
+}
+
+/**
+ * bd_fs_exfat_info_copy: (skip)
+ *
+ * Creates a new copy of @data.
+ */
+BDFSExfatInfo* bd_fs_exfat_info_copy (BDFSExfatInfo *data) {
+    if (data == NULL)
+        return NULL;
+
+    BDFSExfatInfo *ret = g_new0 (BDFSExfatInfo, 1);
+
+    ret->label = g_strdup (data->label);
+    ret->uuid = g_strdup (data->uuid);
+    ret->sector_size = data->sector_size;
+    ret->sector_count = data->sector_count;
+    ret->free_sectors = data->free_sectors;
+    ret->cluster_size = data->cluster_size;
+    ret->cluster_count = data->cluster_count;
+    ret->free_clusters = data->free_clusters;
+
+    return ret;
+}
+
+/**
+ * bd_fs_exfat_info_free: (skip)
+ *
+ * Frees @data.
+ */
+void bd_fs_exfat_info_free (BDFSExfatInfo *data) {
+    if (data == NULL)
+        return;
+
+    g_free (data->label);
+    g_free (data->uuid);
+    g_free (data);
+}
+
+/**
+ * bd_fs_exfat_mkfs:
+ * @device: the device to create a new exfat fs on
+ * @extra: (allow-none) (array zero-terminated=1): extra options for the creation (right now
+ *                                                 passed to the 'mkexfat' utility)
+ * @error: (out): place to store error (if any)
+ *
+ * Returns: whether a new exfat fs was successfully created on @device or not
+ *
+ * Tech category: %BD_FS_TECH_EXFAT-%BD_FS_TECH_MODE_MKFS
+ */
+gboolean bd_fs_exfat_mkfs (const gchar *device, const BDExtraArg **extra, GError **error) {
+    const gchar *args[3] = {"mkexfatfs", device, NULL};
```

Comment:
This looks simple enough so that even the other `exfat-utils` should work fine. This is going to be the most used tool and it's good that no fallback is needed at this time.

Reply:
I've changed the mkfs and fsck commands to `mkfs.exfat` and `fsck.exfat`, these should work with both exfat-utils and the options are also compatible for now.

Reply:
Oh, nice!